src/Roslyn.Diagnostics.Analyzers/CSharp/BlankLines/CSharpBlankLinesBetweenStatementsCodeFixProvider.cs

```diff
@@ -0,0 +1,56 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+#nullable disable warnings
+
+using System;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.CSharp;
+using Roslyn.Diagnostics.Analyzers;
+
+namespace Roslyn.Diagnostics.CSharp.Analyzers.BlankLines
+{
+    [ExportCodeFixProvider(LanguageNames.CSharp)]
+    public sealed class CSharpBlankLinesBetweenStatementsCodeFixProvider : CodeFixProvider
+    {
+        private static readonly SyntaxTrivia s_endOfLine = SyntaxFactory.EndOfLine(Environment.NewLine);
+
+        public override ImmutableArray<string> FixableDiagnosticIds
+            => ImmutableArray.Create(RoslynDiagnosticIds.BlankLinesBetweenStatementsRuleId);
+
+        public override FixAllProvider GetFixAllProvider()
+            => WellKnownFixAllProviders.BatchFixer;
+
+        public override Task RegisterCodeFixesAsync(CodeFixContext context)
+        {
+            var document = context.Document;
+            var diagnostic = context.Diagnostics.First();
```

Comment:
We have some code-fixes doing a loop over the `Diagnostics` and some other doing `.First()` what is the right way?

Reply:
https://github.com/dotnet/roslyn/pull/44824#discussion_r439754184

Reply:
It might be worth doing an anlyzer for this then.

Reply:
I think this logic is correct. If there were multiple diagnostics for the same token, we'd only want to process one of them anyways. If we looked at all diagnostics, we'd have to dedupe. So this has the same effect.